src/main/java/ch/njol/skript/conditions/CondIsWhitelisted.java

```diff
@@ -19,72 +19,73 @@
 package ch.njol.skript.conditions;
 
 import org.bukkit.Bukkit;
-import org.bukkit.entity.Player;
+import org.bukkit.OfflinePlayer;
 import org.bukkit.event.Event;
 import org.eclipse.jdt.annotation.Nullable;
 
 import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Condition;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @Name("Is Whitelisted")
-@Description("Whether or not the server or a player is whitelisted.")
-@Examples({"if server is whitelisted:", "if player is whitelisted"})
-@Since("2.5.2")
+@Description("Whether or not the server or a player is whitelisted, or the server is whitelist enforced.")
+@Examples({
+	"if player is whitelisted:",
+	"if server is whitelisted:",
+	"if server is whitelist enforced:"
+})
+@Since("2.5.2, INSERT VERSION (enforce, offline players)")
+@RequiredPlugins("Minecraft 1.17+ (enforce)")
 public class CondIsWhitelisted extends Condition {
-	
+
+	private static final boolean ENFORCE_SUPPORT = Skript.methodExists(Bukkit.class, "isWhitelistEnforced");
+
 	static {
-		Skript.registerCondition(CondIsWhitelisted.class,
-			"[the] server (is|1¦is(n't| not)) white[ ]listed",
-			"%players% (is|are)(|1¦(n't| not)) white[ ]listed");
+		List<String> patterns = new ArrayList<>();
+		patterns.add("[the] server (is|not:(isn't|is not)) white[ ]listed");
+		patterns.add("%offlineplayers% (is|are|not:(isn't|is not|aren't|are not)) white[ ]listed");
+		if (ENFORCE_SUPPORT)
+			patterns.add("[the] white[ ]list (is|not:(isn't|is not)) enforced");
+		Skript.registerCondition(CondIsWhitelisted.class, patterns.toArray(new String[0]));
```

Comment:
Use string concatenation on a single string here. No need for a list.

Reply:
What exactly would that look like to you? I'm having a hard time thinking up a good way to do that, and this approach seems fine to me.

Best I can come up with is 
```
"[the] server (is|not:(isn't|is not)) white[ ]list[ed]" + (ENFORCE_SUPPORT ? " [:enforced]" : ""),
"%offlineplayers% (is|are|not:(isn't|is not|aren't|are not)) white[ ]listed"
```

Reply:
I see that it's multiple patterns and not a single pattern. Single patterns should always be concatenated. In that case you should be using a String array as you will know the defined length of it.
```java
String[] patterns = new String[ENFORCE_SUPPORT ? 3 : 2];
patterns[0] = "[the] server (is|not:(isn't|is not)) white[ ]listed";
patterns[1] = "%offlineplayers% (is|are|not:(isn't|is not|aren't|are not)) white[ ]listed";
if (ENFORCE_SUPPORT)
    patterns[2] = "[the] white[ ]list (is|not:(isn't|is not)) enforced";
```